code/game/world.dm

```diff
@@ -0,0 +1,403 @@
+				if (!line)
+					continue
+				if (copytext(line, 1, 2) == ";")
+					continue
+
+				var/title = "Mentor"
+				var/rights = admin_ranks[title]
+
+				var/ckey = copytext(line, 1, length(line)+1)
+				var/datum/admins/D = new /datum/admins(title, rights, ckey)
+				D.associate(directory[ckey])
+
+/world/proc/update_status()
+	var/s = ""
+
+	if (config && config.server_name)
+		s += "<b>[config.server_name]</b> &#8212; "
+
+	s += "<b>[station_name()]</b>";
+	s += " ("
+	s += "<a href=\"http://\">" //Change this to wherever you want the hub to link to.
+//	s += "[game_version]"
+	s += "Default"  //Replace this with something else. Or ever better, delete it and uncomment the game version.
+	s += "</a>"
+	s += ")"
+
+	var/list/features = list()
+
+	if(SSticker)
+		if(master_storyteller)
+			features += master_storyteller
+	else
+		features += "<b>STARTING</b>"
+
+	if (!config.enter_allowed)
+		features += "closed"
+
+	features += config.abandon_allowed ? "respawn" : "no respawn"
+
+	if (config && config.allow_vote_mode)
+		features += "vote"
+
+	if (config && config.allow_ai)
+		features += "AI allowed"
+
+	var/n = 0
+	for (var/mob/M in GLOB.player_list)
+		if (M.client)
+			n++
+
+	if (n > 1)
+		features += "~[n] players"
+	else if (n > 0)
+		features += "~[n] player"
+
+
+	if (config && config.hostedby)
+		features += "hosted by <b>[config.hostedby]</b>"
+
+	if (features)
+		s += ": [jointext(features, ", ")]"
+
+	/* does this help? I do not know */
+	if (src.status != s)
+		src.status = s
+
+#define FAILED_DB_CONNECTION_CUTOFF 5
+var/failed_db_connections = 0
+var/failed_old_db_connections = 0
+
+/hook/startup/proc/connectDB()
+	if(!setup_database_connection())
+		log_world("Your server failed to establish a connection with the feedback database.")
+	else
+		log_world("Feedback database connection established.")
+	return 1
+
+/proc/setup_database_connection()
+
+	if(failed_db_connections > FAILED_DB_CONNECTION_CUTOFF)	//If it failed to establish a connection more than 5 times in a row, don't bother attempting to conenct anymore.
+		return 0
+
+	if(!dbcon)
+		dbcon = new()
+
+	var/user = sqllogin
+	var/pass = sqlpass
+	var/db = sqldb
+	var/address = sqladdress
+	var/port = sqlport
+
+	dbcon.Connect("dbi:mysql:[db]:[address]:[port]", "[user]", "[pass]")
+	. = dbcon.IsConnected()
+	if ( . )
+		failed_db_connections = 0	//If this connection succeeded, reset the failed connections counter.
+	else
+		failed_db_connections++		//If it failed, increase the failed connections counter.
+		log_world(dbcon.ErrorMsg())
+
+	return .
+
+//This proc ensures that the connection to the feedback database (global variable dbcon) is established
+proc/establish_db_connection()
+	if(failed_db_connections > FAILED_DB_CONNECTION_CUTOFF)
+		return 0
+
+	if(!dbcon || !dbcon.IsConnected())
+		return setup_database_connection()
+	else
+		return 1
+
+/world/proc/incrementMaxZ()
+	maxz++
+	SSmobs.MaxZChanged()
+
+/world/proc/change_fps(new_value = 30)
```

Comment:
I just came here to make sure this is here

Reply:
?